Client/script.js

```diff
@@ -0,0 +1,45 @@
+//Event Listeners
+
+const searchButton = document.getElementById('search')
+const randomButton = document.getElementById('randombutton')
+
+function showResults(e){
+    e.preventDefault();
+    let submitter = e.submitter
+    if (topic != ""){
+        fetch(`http://localhost:8080/topics/${topic}`)
+        .then(resp => resp.text())
+        .then(data => {
+
+        })
+    }
+}
+
+
+
+
+//for js file
+// const searchButton = document.querySelector("#search");
+// searchButton.addEventListener("click", () => {
+//   fetch(`http://localhost:8080/foods`)
+//     .then((res) => res.text())
+//     .then(insertResults);
+//   window.location.href = "results.html";
+// });
+
+// function insertResult(result) {
```

Comment:
Here at line 30, I am not sure where you are getting the result parameter from. Would you instead need the function to be created inside the .then to access the data from the fetch? or maybe on line 26 you need `.then(insertResults(data/results)` and make the function above the event listener?